### ISSUE TITLE
New Request: Better Session Management

### ISSUE THREAD

**Issue by amir00008** [deanoemcke/thegreatsuspender#1119]
Please complete the following information when submitting a feature request or bug report.
 * Extension version: v7.1.0
 * Browser name & version: Brave & Chrome (latest always)
 * Operating system & version: Win 10 and Mac latest

These would be very exciting features:
1) Add Session Management link to the extension tab (easy access)
2) Ability to save "Window" as a session instead of the whole session (much easier to micro-manage each window session)
3) Ability to delete "Window" in the saved session (once we are done with the research or whatever we use it for, it would be nice to delete specific window since we don't need it anymore but retain the session)
4) Ability to update or potentially overwrite the current session (I've already seen a similar request. However, this will give us an ability to restore previous Window/Session and work on it, add or remove tabs, then when saving, overwrite the existing Session .. maybe Window as well, but let's keep it simple. Just update session.)

These feature enhancements would greatly improve the functionality of this app and make other tab session managers obsolete. Thank you for everything.

**Comment by rakitanc** [comment]
It's pretty much all the same feature I'm looking for in the upcoming updates.

I request Dean to please look at it and consider this, each request in this will definitely enhances the user experience and make it more easier that it already is.

**Comment by tyeth** [comment]
Can I poke this and confirm the window-based session management would be useful. I'm also here to see how complex it would be to remember the Chrome tab groups (and colours ideally)

My example of one window I wish to regularly restore/save: 
![image](https://user-images.githubusercontent.com/6692083/80943256-ae046080-8dde-11ea-885b-1019a741856e.png)

**Comment by nfultz** [comment]
@tyeth until Google exposes the tab groups metadata to the `chrome.tabs` API, it isn't possible to control tab groups / colors from an extension.

**Comment by ossilator** [comment]
is it technically possible to use features offered by other extensions? a dependency on session buddy would be a lot more elegant and easier to maintain than the minimalistic built-in session manager.

**Comment by nfultz** [comment]
Yes, there's a way to do cross-extension messaging.  Doesn't look like it  is really used very much though: [search](https://www.google.com/search?q=chrome.runtime.onMessageExternal+site%3Agithub.com).

But it should probably be the other way around, where the session manager asks GS to suspend / wake a tab. For example, I added a feature in [odinochka](http://odinochka.xyz) to un-munge GS URLs of suspended tabs.

Google has also been really slow about approving extensions on the store when you add extra permissions, adding cross-extension messaging would probably add a 3-6 week wait to the release cycle.
